commit 6c16da9834e158e26edca046ddd597f850d9e244
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Thu Sep 28 00:41:02 2017 +0200

    hurd: Fix `seekdir' symbol exposition from `rewinddir'
    
    `seekdir' is MISC || XOPEN, it should not be exposed along `rewinddir' which
    is POSIX.
    
            * include/dirent.h (__seekdir): New declaration.
            * sysdeps/mach/hurd/seekdir.c (seekdir): Rename to __seekdir and
            redefine as weak alias.
            * sysdeps/mach/hurd/rewinddir.c (__rewinddir): Use __seekdir instead
            of seekdir.

Index: glibc-2.26/include/dirent.h
===================================================================
--- glibc-2.26.orig/include/dirent.h
+++ glibc-2.26/include/dirent.h
@@ -48,6 +48,7 @@ extern DIR *__alloc_dir (int fd, bool cl
 			 const struct stat64 *statp)
      internal_function attribute_hidden;
 extern __typeof (rewinddir) __rewinddir;
+extern __typeof (seekdir) __seekdir;
 extern __typeof (dirfd) __dirfd;
 
 extern void __scandir_cancel_handler (void *arg) attribute_hidden;
Index: glibc-2.26/sysdeps/mach/hurd/rewinddir.c
===================================================================
--- glibc-2.26.orig/sysdeps/mach/hurd/rewinddir.c
+++ glibc-2.26/sysdeps/mach/hurd/rewinddir.c
@@ -24,7 +24,7 @@
 void
 __rewinddir (DIR *dirp)
 {
-  seekdir (dirp, (off_t) 0L);
+  __seekdir (dirp, (off_t) 0L);
 }
 libc_hidden_def (__rewinddir)
 weak_alias (__rewinddir, rewinddir)
Index: glibc-2.26/sysdeps/mach/hurd/seekdir.c
===================================================================
--- glibc-2.26.orig/sysdeps/mach/hurd/seekdir.c
+++ glibc-2.26/sysdeps/mach/hurd/seekdir.c
@@ -22,9 +22,8 @@
 #include "dirstream.h"
 
 /* Seek to position POS in DIRP.  */
-/* XXX should be __seekdir ? */
 void
-seekdir (DIR *dirp, long int pos)
+__seekdir (DIR *dirp, long int pos)
 {
   __libc_lock_lock (dirp->__lock);
   /* Change our entry index pointer to POS and discard any data already
@@ -35,3 +34,5 @@ seekdir (DIR *dirp, long int pos)
   dirp->__size = 0;
   __libc_lock_unlock (dirp->__lock);
 }
+
+weak_alias (__seekdir, seekdir)
